mcs/class/System/System.Net.Sockets/Socket.cs

```diff
@@ -1375,7 +1454,7 @@ public IAsyncResult BeginConnect (EndPoint end_point, AsyncCallback callback, ob
 			is_bound = false;
 			connect_in_progress = true;
 
-			socket_pool_queue (SocketAsyncWorker.Dispatcher, sockares);
+			IOSelector.Add (sockares.handle, new IOSelectorJob (IOOperation.Write, s => BeginConnectCallback ((SocketAsyncResult) s), sockares));
```

Comment:
we pass the directly to `IOSelector`, as `IOSelector` is not specific to `SocketAsyncResult`, so we can also use it with `ProcessAsyncReader`, and potentially in the future for async file operations. It uses `IOAsyncResult` internally which has no clue of the handle, but is a simple implementation of `IAsyncResult`.